frontend/src/components/pages/rp-connect/secrets/Secrets.List.tsx

```diff
@@ -0,0 +1,172 @@
+import {Link} from 'react-router-dom';
+import {PencilIcon, TrashIcon} from '@heroicons/react/outline';
+import EmptyConnectors from '../../../../assets/redpanda/EmptyConnectors.svg';
+import {DeleteSecretRequest, Secret} from '../../../../protogen/redpanda/api/dataplane/v1alpha2/secret_pb';
+
+const {ToastContainer, toast} = createStandaloneToast();
+
+const CreateSecretButton = () => {
+    return (<Box style={{display: 'flex', marginBottom: '.5em'}}>
+        <Link to={'/rp-connect/secret/create'}><Button variant="outline" colorScheme="brand">Create Secret</Button></Link>
+    </Box>)
+}
+
+const EmptyPlaceholder = () => {
+    return <Flex alignItems="center" justifyContent="center" flexDirection="column" gap="4" mb="4">
+        <Image src={EmptyConnectors}/>
+        <Box>You have no Redpanda Connect secrets.</Box>
+        <CreateSecretButton/>
+    </Flex>
+};
+
+@observer
+class RpConnectSecretsList extends PageComponent {
+
+    initPage(p: PageInitHelper) {
+        p.addBreadcrumb('Redpanda Connect Secret Manager', '/rp-connect/secrets');
+        this.refreshData(true);
+        appGlobal.onRefresh = () => this.refreshData(true);
+    }
+
+    refreshData(force: boolean) {
+        if (!Features.pipelinesApi) return;
+
+        rpcnSecretManagerApi.refreshSecrets(force)
+            .catch((err) => {
+                if (String(err).includes('404')) {
+                    // Hacky special handling for OSS version, it is possible for the /endpoints request to not complete in time for this to render
+                    // so in this case there would be an error shown because we were too fast (with rendering, or the req was too slow)
+                    // We don't want to show an error in that case
+                    return;
+                }
+
+                if (Features.pipelinesApi) {
+                    toast({
+                        status: 'error', duration: null, isClosable: true,
+                        title: 'Failed to load pipelines',
+                        description: String(err),
+                    });
+                }
+            })
+    }
+
+    async deleteSecret(id: string) {
+        await rpcnSecretManagerApi.delete(new DeleteSecretRequest({
+            id
+        }))
+        this.refreshData(true);
+    }
+
+    render() {
+
+        const filteredSecrets = (rpcnSecretManagerApi.secrets ?? [])
+            .filter(u => {
+                const filter = uiSettings.rpncSecretList.quickSearch;
+                if (!filter) return true;
+                try {
+                    const quickSearchRegExp = new RegExp(filter, 'i');
+                    if (u.id.match(quickSearchRegExp))
+                        return true;
+                    return false;
+                } catch {
+                    return false;
+                }
+            });
+
+        return (
+            <PageContent>
+                <Section>
+                    <ToastContainer/>
```

Comment:
Do we need ToastContainer copied everywhere? There's no global provider?

Reply:
Again, I use pipeline as model, we can improve it later, when we have all console context